src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -420,23 +484,29 @@ def register_options(cls, register):
              help='Use these javac plugins.')
     register('--javac-plugin-args', advanced=True, type=dict, default={}, fingerprint=True,
              help='Map from javac plugin name to list of arguments for that plugin.')
+    cls.register_jvm_tool(register, 'javac-plugin-dep', classpath=[],
+                          help='Search for javac plugins here, as well as in any '
+                               'explicit dependencies.')
 
     register('--scalac-plugins', advanced=True, type=list, fingerprint=True,
              help='Use these scalac plugins.')
     register('--scalac-plugin-args', advanced=True, type=dict, default={}, fingerprint=True,
              help='Map from scalac plugin name to list of arguments for that plugin.')
-
-    # Scalac plugin jars must already be available at compile time, because they need to be listed
-    # on the scalac command line. We search for available plugins on the tool classpath provided
-    # by //:scalac-plugin-jars.  Therefore any in-repo plugins must be published, so they can be
-    # pulled in as a tool.
-    # TODO: Ability to use built in-repo plugins via their context jars.
-    cls.register_jvm_tool(register, 'scalac-plugin-jars', classpath=[])
+    cls.register_jvm_tool(register, 'scalac-plugin-jars', classpath=[],
+                          removal_version='1.5.0.dev0',
+                          removal_hint='Use --compile-zinc-scalac-plugin-dep instead.')
```

Comment:
Would it make sense to put the new options on https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/jvm/subsystems/scala_platform.py and https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/jvm/subsystems/java.py , rather than putting them on `zinc`?